components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/utils/ServiceCatalogUtils.java

```diff
@@ -85,17 +87,31 @@
     private static String updateServiceUrl(String currentUrl) throws ResolverException {
         /*
             Supported formats
-            https://{host}:{port}/api1
-            https://{url}/api1
+            https://{MI_HOST}:{MI_PORT}/api1
+            https://{MI_URL}/api1
         */
-        SystemResolver resolver = new SystemResolver();
-        if (currentUrl.contains(HOST) && currentUrl.contains(PORT)) {
-            resolver.setVariable(MI_HOST);
-            String host = resolver.resolve();
-            resolver.setVariable(MI_PORT);
-            String port = resolver.resolve();
-            currentUrl = currentUrl.replace(HOST, host).replace(PORT, port);
+        if (currentUrl.contains(HOST) || currentUrl.contains(PORT)) {
+            Map<String, Object> configParser = ConfigParser.getParsedConfigs();
+            String envHost = System.getenv(MI_HOST);
+            // update the {MI_HOST} with server.hostname value
+            if (currentUrl.contains(HOST) && (envHost == null || envHost.isEmpty())) {
+                envHost = (String) configParser.get(SERVER_HOSTNAME);
+            }
+            String envPort = System.getenv(MI_PORT);
```

Comment:
Let's not resolve these per artifact. Let's keep 4 pre-resolved hosts for http and https (for both usages of host+port and url) and used them here based on whether the url is parameterized.

Reply:
Fixed

Reply:
The fact that I am saying is we don't need to do these per artifact. We have have simply have 4 resolved values, MI_ HOST,  HTTP_MI_URL, HTTP_PORT, HTTPS_PORT and replace them depending on the usage.

Reply:
Done